phoenix-core/src/main/java/org/apache/hadoop/hbase/regionserver/IndexHalfStoreFileReaderGenerator.java

```diff
@@ -395,7 +395,10 @@ public KeyValueScanner preStoreScannerOpen(final ObserverContext<RegionCoprocess
                         .getReader().getHFileReader().hasMVCCInfo(), readPt));
             }
         }
-        keyValueScanners.addAll(((HStore) store).memstore.getScanners(readPt));
+        List<KeyValueScanner> memstoreScanners = ((HStore) store).memstore.getScanners(readPt);
```

Comment:
In general, we don't ever want to be casting the interface (`Store`) to the implementation (`HStore`). Is phoenix fundamentally doing something here that requires it to be tied to some specific HStore functionality?

Reply:
Actually i didn't want to touch the other part of code to fix this specific issue.

Reply:
@joshelser Looks like we need HStore to access memstore specifically. We can't access the memstore and get its scanners from the store object.

Reply:
@joshelser - Thanks for pointing that out(my bad i didn't get that in first look at the code ). Updated PR, please have a look.